src/main/java/io/appform/functionmetrics/FunctionTimerAspect.java

```diff
@@ -43,8 +52,35 @@ public Object around(ProceedingJoinPoint joinPoint) throws Throwable {
         final String methodName = Strings.isNullOrEmpty(monitoredFunction.method())
                                     ? callSignature.getName()
                                     : monitoredFunction.method();
-        log.trace("Called for class: {} method: {}", className, methodName);
-        final FunctionInvocation invocation = new FunctionInvocation(className, methodName);
+        final String[] parameterNames = methodSignature.getParameterNames();
+        String parameterString = "";
+        List<String> paramValues = Streams.zip(Arrays.stream(methodSignature.getMethod().getParameters()), Arrays.stream(joinPoint.getArgs()), Pair::new)
+                .filter(pair -> {
+                    MetricTerm metricTerm = pair.getKey().getAnnotation(MetricTerm.class);
+                    return metricTerm != null;
+                })
+                .map(pair -> {
+                    MetricTerm metricTerm = pair.getKey().getAnnotation(MetricTerm.class);
+                    return new Pair<>(metricTerm.order(), pair.getValue());
+                })
+                .sorted(Comparator.comparingInt(Pair::getKey))
+                .map(Pair::getValue)
+                .map(paramValue -> paramValue != null ? paramValue : "")
+                .map(paramValue -> paramValue instanceof String ? (String) paramValue : "")
+                .map(String::trim)
+                .map(paramValue -> NON_ALPHABET_CHARS_PATTERN.matcher(paramValue).replaceAll(""))
+                .map(String::toLowerCase)
```

Comment:
After the above series of steps, you might end up with empty strings. Filter them out.

Reply:
Also you might want to use Guava CaseFormat to convert to camel case instead of just all lower to increase readability. Maybe take this as option?

Reply:
> After the above series of steps, you might end up with empty strings. Filter them out.

This is done later. The decision to skip this metric measurement is done after looking at all the parameter values. We don't want to end up with a combinatorially explosive metric term. So currently I skip pushing this measurement even if one of the parameter values is null or empty.

Reply:
> Also you might want to use Guava CaseFormat to convert to camel case instead of just all lower to increase readability. Maybe take this as option?

Yes. This will definitely improve readability. I have added it as an option enabled by default.